src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -112,31 +112,30 @@ void ChipDeviceScanner::Shutdown()
         },
         this);
 
-    mIsInitialized = false;
+    mCurrentState = ChipDeviceScannerState::SCANNER_UNINITIALIZED;
 }
 
 CHIP_ERROR ChipDeviceScanner::StartScan(System::Clock::Timeout timeout)
 {
+    CHIP_ERROR err              = CHIP_NO_ERROR;
+    static bool startingScanner = false; // To prevent function reentrancy
```

Comment:
Why this variable is static? I think that it would be better to keep in the class instance. The same thing in the `StopScan()`.